`bio-orchestrator` misroutes "latent embedding" queries to `scrna-orchestrator` instead of `scrna-embedding` due to longest-match keyword precedence

## Summary

`detect_skill_with_hint_from_query()` in `skills/bio-orchestrator/orchestrator.py` uses a longest-match strategy over `KEYWORD_MAP` as its final routing stage. For the query `Build a latent embedding for this single-cell dataset`, the keyword `single-cell` (11 characters, mapped to `scrna-orchestrator`) has a longer character count than `embedding` (9) or `latent` (6), both of which are mapped to `scrna-embedding`. The longest-match rule therefore routes the query to `scrna-orchestrator`, contradicting the semantic intent of the query, the design of the early-return guards, and the expectation of the existing regression test.

## Evidence

`skills/bio-orchestrator/orchestrator.py`, `KEYWORD_MAP` (lines approximately 68-93):

```python
"latent":       "scrna-embedding",    # 6 chars
"embedding":    "scrna-embedding",    # 9 chars
...
"single-cell":  "scrna-orchestrator", # 11 chars <- wins by character count
```

Routing function (`detect_skill_with_hint_from_query`, lines 321-345):

```python
wants_embedding = any(term in query_lower for term in SCRNA_EMBEDDING_TERMS)
wants_downstream = any(term in query_lower for term in SCRNA_DOWNSTREAM_TERMS)
has_latent_artifact = any(term in query_lower for term in SCRNA_LATENT_ARTIFACT_TERMS)

if has_latent_artifact and wants_downstream:
    return ("scrna-orchestrator", ...)
if wants_embedding and wants_downstream:
    return ("scrna-embedding", ...)
# Falls through to longest-match KEYWORD_MAP scan
best_skill, best_len = None, 0
for keyword, skill in KEYWORD_MAP.items():
    if keyword in query_lower and len(keyword) > best_len:
        best_skill = skill
        best_len = len(keyword)
```

For `build a latent embedding for this single-cell dataset`:

| Signal | Value |
| --- | --- |
| `wants_embedding` | `True` (`latent` and `embedding` match) |
| `wants_downstream` | `False` (no `marker`, `cluster`, `annotation`, etc.) |
| `has_latent_artifact` | `False` (no `integrated h5ad`, `after scvi`) |

Neither early-return guard fires. The longest-match scan selects `single-cell` (11 chars) -> `scrna-orchestrator`.

Failing test (`skills/scrna-embedding/tests/test_scrna_embedding.py`, line 649):

```python
assert module.detect_skill_from_query(
    "Build a latent embedding for this single-cell dataset"
) == "scrna-embedding"
# AssertionError: 'scrna-orchestrator' != 'scrna-embedding'
```

## Root Cause

The early-return guards correctly encode the semantic intent: `wants_embedding` and not `wants_downstream` should route to `scrna-embedding`. However, the code structure requires both `wants_embedding` and `wants_downstream` to be `True` to activate the `scrna-embedding` early return. When only `wants_embedding` is `True`, the pure embedding case falls through to the keyword map, where the generic `single-cell` token outcompetes the more specific `latent` and `embedding` tokens by raw length.

## Proposed Fix

Add a third early-return guard for the pure embedding case:

```python
if wants_embedding and not wants_downstream and not has_latent_artifact:
    return (
        "scrna-embedding",
        "Detected an embedding-focused scRNA workflow. "
        "Run `scrna-embedding` to produce `integrated.h5ad` with a "
        "scVI/scANVI latent space. For downstream clustering and annotation, "
        "run `scrna-orchestrator` with `--use-rep X_scvi` afterwards.",
    )
```

This fires before the `KEYWORD_MAP` scan and correctly handles queries that express a clear embedding intent without downstream signals.

Additionally, extend the routing test suite with ambiguous cases to prevent regressions, for example queries containing `single-cell` plus `latent` but no downstream terms, and queries containing `single-cell` alone.

--- a/skills/bio-orchestrator/orchestrator.py
+++ b/skills/bio-orchestrator/orchestrator.py
@@ -331,13 +331,23 @@ def detect_skill_with_hint_from_query(query: str) -> tuple[str | None, str]:
             "with `--use-rep X_scvi` on `integrated.h5ad` to run clustering, annotation, "
             "and contrastive markers after scVI.",
         )
-    if wants_embedding and wants_downstream:
-        return (
-            "scrna-embedding",
-            "Detected a two-step advanced scRNA workflow. First run `scrna-embedding` to "
-            "produce `integrated.h5ad`, then run `scrna-orchestrator` with "
-            "`--use-rep X_scvi` for downstream clustering, annotation, and contrastive markers.",
-        )
+
+    if wants_embedding:
+        if wants_downstream:
+            return (
+                "scrna-embedding",
+                "Detected a two-step advanced scRNA workflow. First run `scrna-embedding` to "
+                "produce `integrated.h5ad`, then run `scrna-orchestrator` with "
+                "`--use-rep X_scvi` for downstream clustering, annotation, and contrastive markers.",
+            )
+        if not has_latent_artifact:
+            return (
+                "scrna-embedding",
+                "Detected an embedding-focused scRNA workflow. Use `scrna-embedding` to "
+                "produce `integrated.h5ad` with a scVI/scANVI latent space before "
+                "running downstream clustering or annotation.",
+            )
+
     # Prefer longest keyword match to avoid ambiguity (e.g. "variant annotation"
     # should match vcf-annotator, not equity-scorer via "variant" substring)
     best_skill = None

--- a/skills/bio-orchestrator/orchestrator.py
+++ b/skills/bio-orchestrator/orchestrator.py
@@ -324,6 +324,8 @@ def detect_skill_with_hint_from_query(query: str) -> tuple[str | None, str]:
     wants_downstream = any(term in query_lower for term in SCRNA_DOWNSTREAM_TERMS)
     has_latent_artifact = any(term in query_lower for term in SCRNA_LATENT_ARTIFACT_TERMS)
 
+    # Chain-aware scRNA routing favors explicit embedding requests unless the
+    # user is clearly asking for downstream analysis on an existing latent artifact.
     if has_latent_artifact and wants_downstream:
         return (
             "scrna-orchestrator",
@@ -331,22 +333,27 @@ def detect_skill_with_hint_from_query(query: str) -> tuple[str | None, str]:
             "with `--use-rep X_scvi` on `integrated.h5ad` to run clustering, annotation, "
             "and contrastive markers after scVI.",
         )
-
+    if wants_embedding and wants_downstream:
+        return (
+            "scrna-embedding",
+            "Detected a two-step advanced scRNA workflow. First run `scrna-embedding` to "
+            "produce `integrated.h5ad`, then run `scrna-orchestrator` with "
+            "`--use-rep X_scvi` for downstream clustering, annotation, and contrastive markers.",
+        )
+    if wants_embedding and has_latent_artifact:
+        return (
+            "scrna-embedding",
+            "Detected an embedding-focused scRNA workflow on an existing latent artifact. "
+            "Use `scrna-embedding` to refresh or rebuild the scVI/scANVI latent space "
+            "before downstream clustering or annotation.",
+        )
     if wants_embedding:
-        if wants_downstream:
-            return (
-                "scrna-embedding",
-                "Detected a two-step advanced scRNA workflow. First run `scrna-embedding` to "
-                "produce `integrated.h5ad`, then run `scrna-orchestrator` with "
-                "`--use-rep X_scvi` for downstream clustering, annotation, and contrastive markers.",
-            )
-        if not has_latent_artifact:
-            return (
-                "scrna-embedding",
-                "Detected an embedding-focused scRNA workflow. Use `scrna-embedding` to "
-                "produce `integrated.h5ad` with a scVI/scANVI latent space before "
-                "running downstream clustering or annotation.",
-            )
+        return (
+            "scrna-embedding",
+            "Detected an embedding-focused scRNA workflow. Use `scrna-embedding` to "
+            "produce `integrated.h5ad` with a scVI/scANVI latent space before "
+            "running downstream clustering or annotation.",
+        )
 
     # Prefer longest keyword match to avoid ambiguity (e.g. "variant annotation"
     # should match vcf-annotator, not equity-scorer via "variant" substring)

--- a/skills/bio-orchestrator/tests/test_orchestrator.py
+++ b/skills/bio-orchestrator/tests/test_orchestrator.py
@@ -153,6 +153,18 @@ class TestDetectSkillWithHint:
         assert skill == "scrna-embedding"
         assert "two-step" in hint.lower() or "scrna-embedding" in hint.lower()
 
+    def test_embedding_without_downstream_prefers_scrna_embedding(self):
+        skill, hint = detect_skill_with_hint_from_query(
+            "Build a latent embedding for this single-cell dataset"
+        )
+        assert skill == "scrna-embedding"
+        assert "embedding-focused" in hint.lower() or "latent space" in hint.lower()
+
+    def test_generic_single_cell_query_still_prefers_orchestrator(self):
+        skill, hint = detect_skill_with_hint_from_query("single-cell dataset")
+        assert skill == "scrna-orchestrator"
+        assert hint == ""
+
 
 # ---------------------------------------------------------------------------
 # detect_multiple_skills

--- a/skills/bio-orchestrator/tests/test_orchestrator.py
+++ b/skills/bio-orchestrator/tests/test_orchestrator.py
@@ -158,10 +158,24 @@ class TestDetectSkillWithHint:
             "Build a latent embedding for this single-cell dataset"
         )
         assert skill == "scrna-embedding"
-        assert "embedding-focused" in hint.lower() or "latent space" in hint.lower()
+        assert "embedding-focused" in hint.lower()
+        assert "latent space" in hint.lower()
+
+    def test_embedding_with_latent_artifact_prefers_scrna_embedding(self):
+        skill, hint = detect_skill_with_hint_from_query(
+            "Run embedding on my integrated.h5ad file"
+        )
+        assert skill == "scrna-embedding"
+        assert "embedding-focused" in hint.lower()
+        assert "latent artifact" in hint.lower()
 
     def test_generic_single_cell_query_still_prefers_orchestrator(self):
         skill, hint = detect_skill_with_hint_from_query("single-cell dataset")
+        assert skill == "scrna-orchestrator"
+        assert hint == ""
+
+    def test_downstream_only_query_still_prefers_orchestrator(self):
+        skill, hint = detect_skill_with_hint_from_query("Cluster my single-cell data")
         assert skill == "scrna-orchestrator"
         assert hint == ""
 

